### PR TITLE
Adding extra args and adding variable to restrict bazel resource

### PR DESCRIPTION
**Pull request by brlathia**
Dockerfile.rocm - Adding gpu target as a args to override this parameter on rocm-ci and installing scripts required by QA
build_rocm_python3 - adding a variable in script, if variable is non empty than it will restrict bazel resource, this change is for build machines on rocm-ci.

Signed-off-by: Brij Lathia <brij.lathia@amd.com>

### CODE REVIEW

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

File: tensorflow/tools/ci_build/Dockerfile.rocm [hunk 1]
```diff
@@ -67,7 +69,7 @@ RUN apt-get install -y software-properties-common
 # Install rocm pkgs
 RUN apt-get update --allow-insecure-repositories && \
     DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-unauthenticated \
-    rocm-dev rocm-libs rccl && \
+    rocm-dev rocm-libs rccl miopengemm && \
```
Comment (jayfurmanek) [inline tensorflow/tools/ci_build/Dockerfile.rocm]: Why is this needed? TF doesn't use miopengemm.

File: tensorflow/tools/ci_build/Dockerfile.rocm [hunk 2]
```diff
@@ -80,7 +82,7 @@ ENV PATH="$ROCM_PATH/bin:${PATH}"
 ENV PATH="$OPENCL_ROOT/bin:${PATH}"
 
 # Add target file to help determine which device(s) to build for
-RUN bash -c 'echo -e "gfx900\ngfx906\ngfx908\ngfx90a\ngfx1030" >> ${ROCM_PATH}/bin/target.lst'
+RUN printf '%s\n' > ${ROCM_PATH}/bin/target.lst ${GPU_DEVICE_TARGETS}
```
Comment (jayfurmanek) [inline tensorflow/tools/ci_build/Dockerfile.rocm:85]: This is better, thanks.

File: build_rocm_python3 [hunk 3]
```diff
@@ -29,6 +29,13 @@ TF_PKG_LOC=/tmp/tensorflow_pkg
 rm -f $TF_PKG_LOC/tensorflow*.whl
 
 yes "" | ROCM_PATH=$ROCM_INSTALL_DIR TF_NEED_ROCM=1 PYTHON_BIN_PATH=/usr/bin/python3 ./configure
+# Explicitly define resource constraints on bazel to avoid overload on rocm-ci
+if [ ! -z $2 ]; then
```
Comment (jayfurmanek) [inline build_rocm_python3]: Can we just use `-n` here? The idea here is if there is and second argument ("1", "foo", etc) then we restrict?

You could perhaps generalize it to take in the settings as the second argument. For example, if the argument is "60000/35/70" you can parse that out and set the resources accordingly.

**Review by jayfurmanek — COMMENTED** [review]
This looks ok, Left a few comments. There is a conflict now that needs resolved.

File: build_rocm_python3 (hunk 3 above)
Comment (brlathia) [reply build_rocm_python3]: Provided an -r option to enable restriction

File: tensorflow/tools/ci_build/Dockerfile.rocm (hunk 1 above)
Comment (brlathia) [reply tensorflow/tools/ci_build/Dockerfile.rocm]: removed miopengemm in new patchset

**Review by jayfurmanek — APPROVED** [review]
This looks good now, thanks.
